starky/src/f3g.rs

```diff
@@ -30,15 +34,55 @@ impl F3G {
             dim: 3,
         }
     }
+}
+
+/// Field modulus = 2^64 - 2^32 + 1
+const M: u64 = 0xFFFFFFFF00000001;
+
+/// 2^128 mod M; this is used for conversion of elements into Montgomery representation.
+const R2: u64 = 0xFFFFFFFE00000001;
+
+/// 2^32 root of unity
+const G: u64 = 1753635133440165772;
+
+/// Number of bytes needed to represent field element
+const ELEMENT_BYTES: usize = core::mem::size_of::<u64>();
+
+impl FieldExtension for F3G {
+    const ELEMENT_BYTES: usize = ELEMENT_BYTES;
+    const IS_CANONICAL: bool = false;
+
+    const ZERO: Self = Self {
+        cube: [Fr::ZERO, Fr::ZERO, Fr::ZERO],
+        dim: 1,
+    };
+    const ONE: Self = Self {
+        cube: [Fr::ONE, Fr::ZERO, Fr::ZERO],
+        dim: 1,
+    };
 
```

Comment:
As we have the `one()` function, so that when initialize const `ONE`,  just invoke the function as:
` const ZERO: Self = Self::zero();`

Reply:
the definition of zero() is not equal  to the definition of ZERO which has a difference in the property of dim( zero().dim=3 and ZERO.dim=1 )

Reply:
As both of `zero()` and `ZERO` stands for the zero of the Field, so we can set  `dim=1` for both of them.  
Then, the previous comment can work fine~

Reply:
But it seems that the zero() and ZERO are used in different scenes like the below code:
```
fn is_zero(&self) -> bool {
        match self.dim {
            1 => self._eq(&Self::ZERO),
            _ => self._eq(&Self::zero()),
        }
    }

```

Reply:
correct, actually ZERO is used for the base field, and zero() is used for the extended field. this is a bit confusing here. the root cause is the pil-stark mixed computation between the base field and the extended field. I think we can refactor this confusing design in another PR, with full discussions.

Reply:
OK, I  maintain the current version of code about ZERO.